plugins/ingest/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -449,7 +451,32 @@ public static MetaData fromString(String value) {
                     throw new IllegalArgumentException("no valid metadata field name [" + value + "]");
             }
         }
-
     }
 
+    private class FieldPath {
```

Comment:
Shall I simply pass in sourceAndMetadata and ingestMetadata maps to the constructor?